src/comments.ts

```diff
@@ -0,0 +1,297 @@
+import vscode from 'vscode'
+import { repoInfo } from './git'
+import { gql, mutateGraphQL, queryGraphQL } from './graphql'
+import { log } from './log'
+
+export function activateComments(context: vscode.ExtensionContext): void {
+    if (!vscode.workspace.getConfiguration('sourcegraph').get<string>('discussionsPreviewEnabled')) {
+        return
+    }
+    log.appendLine('Discussions preview is enabled')
+    const commentProvider = new CommentProvider()
+    context.subscriptions.push(vscode.workspace.registerDocumentCommentProvider(commentProvider))
+}
+
+class CommentProvider implements vscode.DocumentCommentProvider {
+    public async provideDocumentComments(
+        document: vscode.TextDocument,
+        token: vscode.CancellationToken
+    ): Promise<vscode.CommentInfo> {
+        let threads: vscode.CommentThread[] = []
+        try {
+            threads = await provideDocumentComments(document)
+        } catch (e) {
+            log.appendLine(`provideDocumentComments ${e}`)
+        }
+
+        const lastLine = document.lineCount - 1
+        const commentingRanges = [new vscode.Range(0, 0, lastLine, document.lineAt(lastLine).range.end.character)]
```

Comment:
Add a comment indicating whether or not these ranges are  _where you can comment_ or _which ranges have a comment_. It seems like the former, but I'm not sure. Is it?

Reply:
added a comment